alacritty/src/input.rs

```diff
@@ -677,12 +677,18 @@ impl<'a, T: EventListener, A: ActionContext<T>> Processor<'a, T, A> {
     /// for its action to be executed.
     fn process_mouse_bindings(&mut self, button: MouseButton) {
         let mods = *self.ctx.modifiers();
+        let mode = *self.ctx.terminal().mode();
+        let mouse_mode = mode.intersects(TermMode::MOUSE_MODE);
 
         for i in 0..self.ctx.config().ui_config.mouse_bindings.len() {
-            let binding = &self.ctx.config().ui_config.mouse_bindings[i];
+            let mut binding = self.ctx.config().ui_config.mouse_bindings[i].clone();
 
-            if binding.is_triggered_by(*self.ctx.terminal().mode(), mods, &button) {
-                let binding = binding.clone();
+            // Require shift for all modifiers when mouse mode is active
+            if mouse_mode {
+                binding.mods |= ModifiersState::SHIFT;
+            }
+
+            if binding.is_triggered_by(mode, mods, &button) {
                 binding.execute(&mut self.ctx);
             }
```

Comment:
Worth to note this behavior in config file. Imagine:

```yaml
mouse_bindings:
  - { mouse: Middle, action: PasteSelection }
  - { mouse: Middle, mods: Shift, action: Paste }  # added by user
```

Alternatively <kbd>Shift</kbd> can be reserved for adjusting selection region.

Reply:
I've added some documentation to the `alacritty.yml` which should make it clear that the <kbd>Shift</kbd> modifier is automatically added in mouse mode.

Could you verify if that comment makes sense for the uninitiated and would actually help against making this mistake @ony?

Also @nixpulvis if you have any comment about the added documentation I'd love to hear it too.